src/policy/ephemeral_policy.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include<policy/ephemeral_policy.h>
+#include<policy/policy.h>
+
+bool CheckValidEphemeralTx(const CTransaction& tx, CFeeRate dust_relay_fee, CAmount txfee, TxValidationState& state)
+{
+    bool has_dust = false;
+    for (const CTxOut& txout : tx.vout) {
+        if (IsDust(txout, dust_relay_fee)) {
+            // We only allow a single dusty output
+            if (has_dust) {
+                return state.Invalid(TxValidationResult::TX_NOT_STANDARD, "dust");
+            }
+            has_dust = true;
+         }
+    }
+
+    // No dust; it's complete standard already
+    if (!has_dust) return true;
+
+    // We never want to give incentives to mine this alone
+    if (txfee != 0) {
+        return state.Invalid(TxValidationResult::TX_NOT_STANDARD, "dust");
+    }
+
+    return true;
+}
+
+std::optional<Txid> CheckEphemeralSpends(const Package& package, CFeeRate dust_relay_rate)
+{
+    // Package is topologically sorted, and PreChecks ensures that
+    // there is up to one dust output per tx.
+
+    assert(std::all_of(package.cbegin(), package.cend(), [](const auto& tx){return tx != nullptr;}));
+
+    // If a parent tx has dust, we have to check for the spend by the child
+    // Maps Txid of parent to output index of the dust.
+    // Single dust per tx possible
+    std::map<Txid, uint32_t> map_tx_dust;
+
+    for (const auto& tx : package) {
+        std::unordered_set<Txid, SaltedTxidHasher> child_unspent_dust;
+        for (const auto& tx_input : tx->vin) {
+            // Parent tx had dust, child MUST be sweeping it
+            // if it's spending any output from parent
```

Comment:
Could comment that `tx_input.prevout.n` doesn't need to equal `map_tx_dust.at(tx_input.prevout.hash)`; a child can spend multiple outputs of a parent.

Reply:
elaborated a bit, let me know if it helps